debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/PrintObjectFieldsHelper.java

```diff
@@ -0,0 +1,321 @@
+			nestedClassHierarchy = valueTypeHelper.findNestedClassHierarchy(localClazz, nestingHierarchy);
+			J9ClassPointer clazz = null;
+
+			U32 flags = new U32(J9VM_FIELD_OFFSET_WALK_INCLUDE_INSTANCE | J9VM_FIELD_OFFSET_WALK_INCLUDE_HIDDEN);
+			for (int i = 0; i < nestedClassHierarchy.length - 1; i++) {
+				clazz = nestedClassHierarchy[i];
+				depth = J9ClassHelper.classDepth(clazz).longValue();
+				boolean found = false;
+				for (superclassIndex = 0; (superclassIndex <= depth) && !found; superclassIndex++) {
+					J9ClassPointer superclass;
+					if (superclassIndex == depth) {
+						superclass = clazz;
+					} else {
+						superclass = J9ClassPointer.cast(clazz.superclasses().at(superclassIndex));
+					}
+					Iterator<J9ObjectFieldOffset> iterator = J9ObjectFieldOffsetIterator.J9ObjectFieldOffsetIteratorFor(superclass.romClass(), clazz, previousSuperclass, flags);
+
+					while (iterator.hasNext()) {
+						J9ObjectFieldOffset result = iterator.next();
+						if (result.getName().equals(nestingHierarchy[i])) {
+							nestedFieldOffset = nestedFieldOffset.add(result.getOffsetOrAddress());
+							found = true;
+							break;
+						}
+					}
+					previousSuperclass = superclass;
+				}
+			}
+
+			localClazz = nestedClassHierarchy[nestedClassHierarchy.length - 1];
+			dataStart = dataStart.add(nestedFieldOffset);
+			previousSuperclass = J9ClassPointer.NULL;
+		}
+		if (showObjectHeader) {
+			if (flatObject) {
+				padding(out, tabLevel);
+				out.format("// EA = %s, offset in top level container = %d;%n", dataStart.getHexAddress(), nestedFieldOffset.add(ObjectModel.getHeaderSize(localObject)).intValue());
+			}
+
+			/* print individual fields */
+			J9UTF8Pointer classNameUTF = localClazz.romClass().className();
+			padding(out, tabLevel);
+			out.format("struct J9Class* clazz = !j9class 0x%X // %s%n", localClazz.getAddress(), J9UTF8Helper.stringValue(classNameUTF));
+			padding(out, tabLevel);
+			out.format("Object flags = %s;%n", J9ObjectHelper.flags(localObject).getHexValue());
+		}
+
+		depth = J9ClassHelper.classDepth(localClazz).longValue();
+		for (superclassIndex = 0; superclassIndex <= depth; superclassIndex++) {
+			J9ClassPointer superclass;
+			if (superclassIndex == depth) {
+				superclass = localClazz;
+			} else {
+				superclass = J9ClassPointer.cast(localClazz.superclasses().at(superclassIndex));
+			}
+
+			U32 flags = new U32(J9VM_FIELD_OFFSET_WALK_INCLUDE_INSTANCE | J9VM_FIELD_OFFSET_WALK_INCLUDE_HIDDEN);
+			Iterator<J9ObjectFieldOffset> iterator = J9ObjectFieldOffsetIterator.J9ObjectFieldOffsetIteratorFor(superclass.romClass(), localClazz, previousSuperclass, flags);
+
+			while (iterator.hasNext()) {
+				J9ObjectFieldOffset result = iterator.next();
+				boolean printField = true;
+				boolean isHiddenField = result.isHidden();
+				boolean isLockword = isHiddenField && (result.getOffsetOrAddress().add(J9Object.SIZEOF).eq(superclass.lockOffset()));
```

Comment:
Please remove the redundant (outermost) parentheses from the right operand of `&&`.